packages/server/src/login/login.controller.ts

```diff
@@ -36,68 +40,55 @@ export class LoginController {
   ): Promise<KhouryRedirectResponse> {
     let user: UserModel;
     user = await UserModel.findOne({
-      where: { username: body.username },
+      where: { email: body.email },
       relations: ['courses'],
     });
 
     if (!user) {
       user = await UserModel.create({ courses: [] });
     }
 
+    // Q: Do we need this if it's not going to change?
     user = Object.assign(user, {
-      username: body.username,
       email: body.email,
       name: body.first_name + body.last_name,
       photoURL: body.photo_url, // We'll have to find away around this becuase the photo urls expire
     });
     await user.save();
 
-    // TODO: Check performanace on this, right now there are mad db queries in here
-    async function courseNameToUserCourse(
-      name: string,
-      role: Role,
-    ): Promise<UserCourseModel> {
-      let course;
-      course = await CourseModel.findOne({
-        where: { name: name },
-      });
-      if (!course) {
-        course = CourseModel.create({ name: name }); // TODO: Figure out how we want to specify the semester and add that
-        await course.save();
-      }
-      let userCourse: UserCourseModel;
-      userCourse = await UserCourseModel.findOne({
-        where: { userId: user.id, courseId: course.id, role: role },
-      });
-      if (!userCourse) {
-        userCourse = await UserCourseModel.create({
-          userId: user.id,
-          course: course,
-          role: role,
-        }).save();
-      }
-      return userCourse;
-    }
-
     const userCourses = [];
     await Promise.all(
-      body.courses
-        .filter((c) => !c.withdraw)
-        .map(async (c) => {
-          const userCourse = await courseNameToUserCourse(
-            c.course_name,
+      body.courses.map(async (c: KhouryStudentCourse) => { 
+        const course: CourseModel = await this.loginCourseService.courseSectionToCourse(c.course, c.section);
+
+        if (course) {
+          const userCourse = await this.loginCourseService.courseToUserCourse(
+            user.id,
+            course.id,
             Role.STUDENT,
           );
           userCourses.push(userCourse);
-        }),
+        }
+      }),
     );
+    
     await Promise.all(
-      body.ta_courses
-        .filter((c) => !c.withdraw)
-        .map(async (c) => {
-          const taCourse = await courseNameToUserCourse(c.course_name, Role.TA);
+      body.ta_courses.map(async (c: KhouryTACourse) => { 
+        // Query for all the courses which match the name of the generic course from Khoury
+        const courses = await getRepository(CourseModel)
```

Comment:
Would it make sense to use the `CourseSectionMappingModel`?
Like:
``` 
await CourseSectionMappingModel.find({
  where: { genericCourseName: courseName },
  relations: ['course'],
});
```

Reply:
Absolutely